### DISCUSSION TITLE
Migration to .net8.0 should I use CoreWCF.Primitives or System.ServiceModel.Primitives

### DISCUSSION THREAD

**Discussion by j-bernhard** [CoreWCF/CoreWCF#1387] [General]
Hy CoreWCF Team / Community, 

I try to upgrade a .net48 to .net8.0, which is using WCF on server and client side. The contract is separated in a own assembly, that includes the services contracts and also the data classes for the services. So that the server and client can use them in there c# projects. 
I see that CoreWCF has own primitives,  for example the ServiceContract - Attribute. But also the package System.ServiceModel.Primitives (8.0.0) provides them. Should I go with the CoreWCF primitives or can I also go with the Micorosoft library? 

Thanks, 
Joerg

**Comment by mconnew** [comment]
System.ServiceModel.* is client, CoreWCF.* is server.

**Reply by mconnew** [reply]
That will teach me to fully read posts. The client System.ServiceModel.* contract attributes are recognized by CoreWCF. If you want a shared library consumed by the client and server, use that. The only downside is you pull in the client libraries into your app so makes your app a bit bigger.

**Reply by j-bernhard** [reply]
@mconnew that is what I hopped to hear. Thx for answer.
